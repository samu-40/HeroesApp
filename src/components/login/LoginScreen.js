import React, { useContext } from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';
import './style.css';

const LoginScreen = ({ history }) => {
    
    const {dispatch} = useContext(AuthContext);

    const handleLogin = () => {
        
        const lastPath = localStorage.getItem('lastPath') || '/';
        const lastSearch = localStorage.getItem('lastSearch');
        
        dispatch({
            type: types.login,
            payload: {name: 'Samuel'},
        });

        history.replace(lastPath + lastSearch);
    };

    return (
        <div className='box'>
            <h1>Login</h1>

            <button 
                className='btn'
                onClick={handleLogin}
            >
                Login
            </button>
        </div>
    )
}

export default LoginScreen;
