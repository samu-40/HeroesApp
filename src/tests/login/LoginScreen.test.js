import '../../setupTest';
import { mount } from "enzyme";
import LoginScreen from '../../components/login/LoginScreen';
import { AuthContext } from '../../auth/AuthContext';

describe('Pruebas en LoginScreen.js', () => {

    const history = {
        replace: jest.fn()
    };

    const contextValue = {
        dispatch: jest.fn()
    };

    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <LoginScreen history={history}/>
        </AuthContext.Provider>
    );

    test('Mostrar correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe llamar al dispatch y al replace', () => {
        
        wrapper.find('button').prop('onClick')();

        expect(contextValue.dispatch).toHaveBeenCalled();
        expect(history.replace).toHaveBeenCalled();

    });
    
    
});
