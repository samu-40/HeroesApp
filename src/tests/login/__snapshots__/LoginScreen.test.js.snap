// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en LoginScreen.js Mostrar correctamente 1`] = `
<div
  className="box"
>
  <h1>
    Login
  </h1>
  <button
    className="btn"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
